test/e2e/exec_test.go

```diff
@@ -548,4 +548,20 @@ RUN useradd -u 1000 auser`, fedoraMinimal)
 		session.WaitWithDefaultTimeout()
 		Expect(session.OutputToString()).To(Not(ContainSubstring(secretsString)))
 	})
+
+	It("podman exec --wait 2 seconds on bogus container", func() {
+		session := podmanTest.Podman([]string{"exec", "--wait", "2", "1234"})
+		session.WaitWithDefaultTimeout()
+		Expect(session).Should(Exit(125))
+		Expect(session.ErrorToString()).To(ContainSubstring("timed out waiting for container to run: 1234"))
+	})
+
+	It("podman exec --wait 5 seconds for started container", func() {
+		session := podmanTest.Podman([]string{"exec", "--wait", "5", "waitCtr", "whoami"})
+
+		session = podmanTest.Podman([]string{"run", "-d", "--name", "waitCtr", ALPINE, "top"})
+		session.Wait(6)
+		Expect(session).Should(Exit(0))
```

Comment:
This has to use another variable for the session here.
Right no this never checks anything with the exec command calling wait() and checking the exit code is important.

Also the wait(6) must be on the exec command not the run command.
I also recommend you check the output of the exec command in addition of the exit code. Only doing exit code checks is not that useful.

Reply:
Updated. PTAL. Also, I replaced ContainSubstring() with Equal() in the first test, as that looked more specific.